moveit_ros/moveit_servo/src/servo_node.cpp

```diff
@@ -162,6 +162,9 @@ void ServoNode::pauseServo(const std::shared_ptr<std_srvs::srv::SetBool::Request
   }
   else
   {
+    // Reset the smoother with the robot's current state in case the robot moved between pausing and unpausing.
+    servo_->resetSmoother(servo_->getCurrentRobotState());
```

Comment:
this is the actual fix, the rest is reducing code duplication

Reply:
Is `pauseServo` always called when switching to another mode? in other words, is it possible to move the arm with planners, while servo is active?

Reply:
Yes, this is possible because both the planning pipelines and servo are separate nodes and can be blasting out data to the same ros2_controller.

However, in most applications (MoveIt Studio very much included), we never have unpaused Servo and motion planning + execution happening at the same time.

I'd call this case "user error".